nomad/vault_test.go

```diff
@@ -528,6 +530,49 @@ func TestVaultClient_RenewalLoop(t *testing.T) {
 	if ttl == 0 {
 		t.Fatalf("token renewal failed; ttl %v", ttl)
 	}
+
+	if client.currentExpiration.Before(time.Now()) {
+		t.Fatalf("found current expiration to be in past %s", time.Until(client.currentExpiration))
```

Comment:
This added check to ensure that after the loop, we track the right expiration.  In my earlier attempt, currentExpiration was an past value that caused renewal loop to renew at every loop without any pausing.